### ISSUE TITLE
[deck] Project Dashboard not showing cluster information after upgrade to 1.10.0

### ISSUE THREAD

**Issue by jfillo** [spinnaker/spinnaker#3490]
### Issue Summary:

Since upgrading to Spinnaker 1.10.0 my Project Dashboard view returns 'No clusters found for selection regions/ namespaces'

### Cloud Provider(s):

Kubernetes v2

### Environment:

Spinnaker 1.10.0 installed via Halyard

### Feature Area: 

deck
clouddriver

### Steps to Reproduce:

Create a Project Dashboard
Under Applications, add an application that has existing ReplicaSets and Pods
Under Clusters, check 'All', choose your Account, and put * as the values for Stack and Detail
Click Save
In the dashboard view you should see the error 'No clusters found for selected region/namespaces'

**Comment by chinaboard** [comment]
+1

**Comment by DmitrySolodovnyk** [comment]
+1 Same problem. I have legacy and v2 accounts for the same k8s cluster in parallel. Legacy works fine, v2 doesn't display running applications.

**Comment by bvanosten** [comment]
+1

**Comment by silvez-dh** [comment]
1+ here, same as the others described

**Comment by Gephorian** [comment]
+1

**Comment by danieldides** [comment]
+1. Rolled my cluster back to 1.9.6 and the cluster immediately populated. Interestingly it looked like it may actually have done the deployment anyway on 1.10, even though the webui was not showing it.

**Comment by kskewes** [comment]
This issue is also present in the pre-release version `release-1.11.x-latest-validated`.

**Comment by kskewes** [comment]
Looking in 1.11 validated logs.
After clicking on Projects in Deck I see:
- Gate, Orca and Clouddriver all log outbound GET requests to Front50. They also log a 200 return.
- Front50 doesn't seem to log anything during this exchange however which is surprising.

eg - Clouddriver logs:
```
2018-12-13 18:36:21.482  INFO 1 --- [.0-7002-exec-10] c.n.s.c.core.services.Front50Service     : ---> HTTP GET http://spin-front50.spinnaker:8080/v2/projects/eeva
<nothing from front50 logged in and around this time>
2018-12-13 18:36:21.626  INFO 1 --- [.0-7002-exec-10] c.n.s.c.core.services.Front50Service     : <--- HTTP 200 http://spin-front50.spinnaker:8080/v2/projects/eeva (144ms)
```

I have logs in Elastic and follow timeline with Kibana.

**Comment by maggieneterval** [comment]
Thanks for the heads up @ezimanyi, I will look into this

**Comment by maggieneterval** [comment]
I think this is a Clouddriver issue, based on a quick git bisect it looks like `project/${projectName}/clusters` stopped returning cluster information when the caching agent was introduced [here](https://github.com/spinnaker/clouddriver/pull/2949/commits/cf02f8f412ede1ecf70e5a36685755c3842f073d) -- @robzienert any idea what might be happening? Thanks!

**Comment by robzienert** [comment]
I'm MIA the next two days, but I'll take a look on Friday. The behavior should have stayed the same if you don't configure any projects to be handled by the caching agent. I haven't seen this internally; I suspect it's a bug on the particular halyard release as opposed to master?

**Comment by maggieneterval** [comment]
This fix will go out with the next patch release -- 1.11.4.

**Comment by beyond-code-github** [comment]
I've just pulled in 1.11.6 and am still seeing this issue

**Comment by DmitrySolodovnyk** [comment]
It is partly solved in 1.11.5. 
Instead of resolving images it put a #0 with a link to a ci that is not configured (/nulljob//0/). 

![screenshot from 2019-01-15 16-48-25](https://user-images.githubusercontent.com/1533610/51188056-d6183980-18e5-11e9-8740-ad46014eca10.png)

**Comment by maggieneterval** [comment]
Sorry about that, thanks for catching -- will look into the image resolution issue.

**Comment by jfillo** [comment]
I am not sure if this is a related issue or not, but I've noticed that with the kubernetes v2 provider I am unable to filter the Project Application Status view by Stack or Detail. I only ever get results if I put in `*` for both values.

**Comment by DmitrySolodovnyk** [comment]
> I am not sure if this is a related issue or not, but I've noticed that with the kubernetes v2 provider I am unable to filter the Project Application Status view by Stack or Detail. I only ever get results if I put in `*` for both values.

Yes, I have the same problem for v2.

**Comment by maggieneterval** [comment]
Can definitely look into that as well, thank you for reporting and confirming @jfillo and @DmitrySolodovnyk!

**Comment by DmitrySolodovnyk** [comment]
1.12.0. Image resolution work fine. 
As notices by @jfillo, filters still don't work. It is important when you have different environments, say staging/test/demo in different namespaces of the same cluster. Might happen that those environments have different app versions, project dashboard would show a single row with an image version that doesn't make any sense in this case. I'd like to use filters to split those envs into different views.
![filters](https://user-images.githubusercontent.com/1533610/52044176-2a5c2400-254a-11e9-9350-77c8e7ce2a39.png)

**Comment by DmitrySolodovnyk** [comment]
Hi @lwander,  From UI perspective we could have another filter by namespace so user can filter clusters view by the same approach we have in k8s. For my case V1 works fine and it looks like a bug in clouddriver. 
![screenshot from 2019-02-01 08-52-30](https://user-images.githubusercontent.com/1533610/52107938-0576b800-2601-11e9-8086-590871e6cebd.png)

**Comment by beyond-code-github** [comment]
> I've just pulled in 1.11.6 and am still seeing this issue

Was this resolved in the end?

**Comment by DmitrySolodovnyk** [comment]
> Hi @lwander, From UI perspective we could have another filter by namespace so user can filter clusters view by the same approach we have in k8s. For my case V1 works fine and it looks like a bug in clouddriver.
> 
@jfillo
Finally I've found the way filter works. Filtering is done by  moniker.spinnaker.io/cluster name, not by moniker.spinnaker.io/stack. If cluster name is something-stg (there is a naming convention), filer by stg stack will work.

**Comment by beyond-code-github** [comment]
I am still seeing this problem. So far I have not had any responses to any of my comments here, should I open a new issue?

**Comment by DmitrySolodovnyk** [comment]
@beyond-code-github, if there are no filters by stages/details project (* as values) dashboards work for me on 1.12.1. 
The filters problem is another clouddriver bug, probably should be handled as a separate issue.

**Comment by beyond-code-github** [comment]
Thanks @DmitrySolodovnyk, I will triple check and then proceed accordingly

**Comment by yemazon** [comment]
+1 Still see this error in 1.12.9
![image](https://user-images.githubusercontent.com/22063951/56621654-1fadaa00-65e2-11e9-8438-00e11e33bb9e.png)

**Comment by jfillo** [comment]
I still can't get this to work. This make the Project page practically useful for those of us using the Kubernetes V2 provider.

**Comment by kskewes** [comment]
We're on 1.15.3 and it has been working for us since somewhere in the 1.13's (IIRC) after a PR by Maggie?
I can't see anything unusual about our setup.
We have only just enabled fiat as well and still ok.
If need a 'next call' after 'failure at xyz' or something please let me know.

**Comment by mkj28** [comment]
Is there another ticket tracking this? Seeing same in 1.20.4.

Intermittently some clusters would show expected data, but mostly seeing "No clusters found for selected region/namespaces"

**Comment by HaroonSaid** [comment]
We use ECS and can't seem to get the project page to display 
using 1.21.0

**Comment by dervistalha** [comment]
the problem is still exist.
![image](https://user-images.githubusercontent.com/71238946/97665501-9a376f80-1a8c-11eb-8778-9e506d9d2240.png)

**Comment by kskewes** [comment]
Ours has stopped working and I think it corresponds with the removal of `stack` and `detail` Moniker key/values from our pipelines. 1.23.0

**Comment by maggieneterval** [comment]
Hi @dervistalha & @kskewes! Trying to determine if what you're seeing is a 1.23.0 regression. If possible, would be great if you could open a fresh issue that includes Clouddriver logs. Thanks!
